### PR TITLE
Upload images fullstack

### PR DESCRIPTION
**Pull request by tunile943**
## Summary

Allow users to upload image to Firebase and store said image URL on Prisma when create/edit event. If no image provided, the display will show default image. Once stored, the image should be displayed appropriately on event card. If a user opts to edit an event without choosing a new image, the event will revert to its previous image.

image

## Testing

We tested by manually creating and editing events, replacing images to see each event update its image appropriately.

### CODE REVIEW

File: frontend/src/components/atoms/Dropzone.tsx [hunk 1]
```diff
@@ -2,15 +2,22 @@ import React, { useState } from "react";
...
  */
-const Dropzone = ({ setError, label }: DropzoneProps) => {
-  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+const Dropzone = ({
+  setError,
+  selectedFile,
+  setSelectedFile,
+  label,
+}: DropzoneProps) => {
+  // const [selectedFile, setSelectedFile] = useState<File | null>(null);
```
Comment (akinfelami) [inline frontend/src/components/atoms/Dropzone.tsx:20]: remove commented out line

File: frontend/src/utils/helpers.ts [hunk 2]
```diff
@@ -169,3 +175,45 @@ export const displayDateInfo = (date: Date) => {
...
   return str[0] + str.substring(1).toLowerCase();
 };
+
+/**
+ * Upload image to firebase
+ * @param userID          - The userID of who uploads the image
+ * @param image           - The image file to upload
+ * @returns The string URL to the event.
+ */
+export const uploadImage = async (userID: string | null, image: File) => {
```
Comment (akinfelami) [inline frontend/src/utils/helpers.ts:185]: use `uploadImageToFirebase` : better naming of functions

File: frontend/src/utils/helpers.ts [hunk 3]
```diff
@@ -169,3 +175,45 @@ export const displayDateInfo = (date: Date) => {
...
+  if (!userID) {
+    console.error("No eventId provided");
```
Comment (akinfelami) [inline frontend/src/utils/helpers.ts:187]: Hmmm how come you are checking if no userID is passed but it errors out as no eventid provided

File: frontend/src/utils/helpers.ts [hunk 4]
```diff
@@ -169,3 +175,45 @@ export const displayDateInfo = (date: Date) => {
...
+  const currentTime = Date.now();
+  const refString = `${userID}_${currentTime}${fileExtension}`;
+  const storage = getStorage();
+  const imageRef = ref(storage, refString);
+  const uploadTask = uploadBytesResumable(imageRef, image);
+
+  // Retrieve the download URL
+  try {
+    // Wait for the upload to complete and retrieve the download URL
+    await uploadTask.then(); // Ensures the upload completes before proceeding
```
Comment (akinfelami) [inline frontend/src/utils/helpers.ts:201]: I'm not sure a `.then()` does anything here if it is already wrapped in a try catch block. Why are you using it?

File: frontend/src/utils/helpers.ts [hunk 5]
```diff
@@ -169,3 +175,45 @@ export const displayDateInfo = (date: Date) => {
...
+    const downloadURL = await getDownloadURL(uploadTask.snapshot.ref);
+    console.log("File available at", downloadURL);
```
Comment (akinfelami) [inline frontend/src/utils/helpers.ts:203]: remove console.log

File: frontend/src/components/organisms/EventForm.tsx [hunk 6]
```diff
@@ -39,7 +44,6 @@ type FormValues = {
   location: string;
   volunteerSignUpCap: string;
   eventDescription: string;
-  eventImage: string;
```
Comment (akinfelami) [inline frontend/src/components/organisms/EventForm.tsx:42]: why did you remove this? Is it no longer needed?

File: frontend/src/components/organisms/EventForm.tsx [hunk 7]
```diff
@@ -169,12 +181,22 @@ const EventForm = ({
           startTime,
           endTime,
         } = data;
+        const userid = await fetchUserIdFromDatabase(user?.email as string);
         const startDateTime = convertToISO(startTime, startDate);
         const endDateTime = convertToISO(endTime, startDate);
+        let imageURL = null;
+        if (typeof eventId === "string") {
+          imageURL = await fetchImageURLFromDB(eventId); // Assign imageURL to previous URL.
+        }
```
Comment (akinfelami) [inline frontend/src/components/organisms/EventForm.tsx:190]: Why do we need a separate query to get the imageUrl from Db if it is already present in the event object?

**Review by akinfelami — CHANGES_REQUESTED** [review]
Hi guys, great work!

I had some questions about the implementation I am not entirely certain about. Let me know what you think!

File: frontend/src/utils/helpers.ts (hunk 3 above)
Comment (tunile943) [reply frontend/src/utils/helpers.ts:187]: Ah sorry, I kept the console error from before we decided to switch to user ID to name the image.

File: frontend/src/components/organisms/EventForm.tsx (hunk 7 above)
Comment (tunile943) [reply frontend/src/components/organisms/EventForm.tsx:190]: Because I could not find a way to update imageURL accordingly to the form in event form. I tried set state, but it does not work properly. And I could not figure out how I could integrate it with dropzone. So I do this workaround instead.

**Review by jasozh — APPROVED** [review]
